packages/blade/src/components/Table/Table.web.tsx

```diff
@@ -204,11 +199,15 @@ const _Table = <Item,>(
     undefined,
   );
   const [hasHoverActions, setHasHoverActions] = React.useState(false);
-  const [VirtualizedTableDimensions, setVirtualizedTableDimensions] = React.useState({
-    width: 0,
-    height: 0,
-  });
-
+  const tableRootComponent = children([]);
+  const isVirtualized = (React.isValidElement<{
+    header?: () => React.ReactElement;
+    children?: React.ReactNode;
+  }>(tableRootComponent) &&
+    Array.isArray(tableRootComponent.props?.children) &&
+    tableRootComponent.props.children?.length &&
+    React.isValidElement<{ children?: React.ReactNode }>(tableRootComponent.props?.children[1]) &&
+    typeof tableRootComponent.props?.children[1]?.props.children === 'function') as boolean;
```

Comment:
This looks a little complex 🙈 Can you add comment on how this checks if table is virualized?

Reply:
refactored this. no jsx walking